streams/src/main/java/org/apache/kafka/streams/state/internals/OffsetCheckpoint.java

```diff
@@ -145,8 +156,14 @@ private void writeEntry(final BufferedWriter writer,
 
                             final String topic = pieces[0];
                             final int partition = Integer.parseInt(pieces[1]);
+                            final TopicPartition tp = new TopicPartition(topic, partition);
                             final long offset = Long.parseLong(pieces[2]);
-                            offsets.put(new TopicPartition(topic, partition), offset);
+                            if (offset >= 0L) {
+                                offsets.put(tp, offset);
+                            } else {
+                                LOG.warn("Read offset={} from checkpoint file for {}", offset, tp);
```

Comment:
Since we may have older checkpoint files that wrote negative offsets, we should just ignore these on read